pages/view/[linkId].tsx

```diff
@@ -3,10 +3,27 @@ import DocumentView from "@/components/view/document-view";
 import { useLink } from "@/lib/swr/use-link";
 import NotFound from "@/pages/404";
 import { useSession } from "next-auth/react";
+import { useState } from "react";
 
 export default function ViewPage() {
   const { link, error } = useLink();
   const { data: session, status } = useSession();
+  const [isArchived, setIsArchived] = useState(false);
```

Comment:
No need to call the API again. See line number 9 `const { link, error } = useLink();` it already calls the `API` using `useSWR()` hook. so all you need to do is include `isArchived` here ` const { expiresAt, emailProtected, password: linkPassword } = link;` and then add the `isArchived` check at this line

```
// If the link is expired, show a 404 page

  if ((expiresAt && new Date(expiresAt) < new Date()) || isArchived) {
    return <NotFound />;
  }
```

Reply:
Thanks a lot for all the feedback. It was a great learning experience for me. I have implemented the fix, the way you described. Please take a look at the latest commit and let me know if any further changes are required.

Reply:
// If the link is expired, show a 404 page modify this comment which basically says // If the link is expired or archived, show a 404 page and you can resolve the comment. Review is done from my end. @mfts will review it too and merge if it make sense. Thanks for your contribution :)

Reply:
Thanks again for the feedback. I have added different error messages. Please take a look and let me know if all the things are correct or if I have to change the way of implementation. I will be happy to work if any additional changes are required.